38-wip/technical_specification.md

```diff
@@ -0,0 +1,72 @@
+# Missing Glue Code for Metadata Ingress Inter-service Communication (Tokay Gecko)
+**Epic Type:** Implementation Epic
+
+## Scope
+
+This epic aims to fill in the missing parts in inter-service communication along the preliminary upload/ingress path provided for 1.0.
+
+### Outline:
+
+#### Metldata Service:
+
+- Implement publisher for deletion and population events
+- Needs to know about MASS and WPS DB and be able to query them
```

Comment:
Looking into foreign databases violates the principles of a microsevice architecture, do we really want to do that?

Instead, the metldata service should use an internal API to request that. Or even better, the metldata service could have its own database of already announced datasets to determine for which datasets deletion events need to be issued.

Actually, the WPS itself also doesn't need to be informed about deletions. It only needs the dataset overview events to know which files belong to which dataset in order to create the work package access token.

However, the claims repository may need to be informed about deletion, so that controlled access grants can be removed.

Reply:
IIRC the bottom line of the video call we had on the subject was that for this temporary implementation we would accept to look "behind the curtain". But actually we have all information already at metldata service anyhow if we fire the events before clearing the database so I guess that would anyway be the easier solution 🤔

Reply:
So, this needs further discussion?

Reply:
@Cito What's the URL of the claims repo and what changes would be needed?